src/wait_list.rs

```diff
@@ -313,20 +313,22 @@ impl<'wait_list, I, O> Borrowed<'wait_list, I, O> {
     }
 
     /// Wake and dequeue the first waiter in the queue, if there is one.
-    pub fn wake_one(&mut self, output: O) {
-        let head = match self.head() {
-            Some(head) => head,
-            None => return,
-        };
+    ///
+    /// Returns ownership of that waiter's input value.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error when there are no wakers in the list.
+    pub fn wake_one(&mut self, output: O) -> Result<I, ()> {
```

Comment:
Might as well name error type, even if it's a `#[non_exhaustive]` unit struct. Helps with readability.

Reply:
An alternative is passing back `O` to the caller. What do you think about `Result<I, WakeOneError<O>>` versus `Result<I, O>`?

Reply:
@udoprog Any thoughts on this?

Reply:
I'd always prefer a named error. It gives the ability to format the message when printed or unwrapped. What data it returns is up to what's needed by users.